wezterm-gui/src/overlay/copy.rs

```diff
@@ -322,14 +322,37 @@ impl CopyRenderable {
     fn select_to_cursor_pos(&mut self) {
         self.clamp_cursor_to_scrollback();
         if let Some(start) = self.start {
-            let start = SelectionCoordinate {
-                x: start.x,
-                y: start.y,
+            let cursor_is_above_start = self.cursor.y < start.y;
+            let start = if matches!(self.selection_mode, SelectionMode::Line) {
```

Comment:
Since `SelectionMode` impls `Eq`, this can be simplified:

```suggestion
            let start = if self.selection_mode == SelectionMode::Line {
```